Analysis/Tasks/PWGHF/HFJpsiCandidateSelector.cxx

```diff
@@ -0,0 +1,257 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFJpsiCandidateSelector.cxx
+/// \brief Jpsi selection task.
+/// \author Biao Zhang <biao.zhang@cern.ch>, CERN
+/// \author Nima Zardoshti <nima.zardoshti@cern.ch>, CERN
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+
+static const int npTBins = 9;
+static const int nCutVars = 4;
+//temporary until 2D array in configurable is solved - then move to json
+//    mass  dcaxy dcaz  pt_e
+constexpr double cuts[npTBins][nCutVars] =
+  {{0.5, 0.2, 0.4, 1},  /* pt<0.5   */
```

Comment:
hi @zhangbiao-phy @nzardosh. In this selection, we have a cut on the max displacement of Jpsi daughter tracks. I think this selection was optimized for a inclusive or prompt jpsi analysis. This strategy would not work for standard HF electron analyses or non prompt Jpsi analyses cause it would cut all the HF component. I am not against keeping this as it is as a first shot, but we need to keep it in mind.